### PR TITLE
ecommerce-start

### PR DESCRIPTION
**Pull request by Sally-Samaan**

### CODE REVIEW

**Comment by Sally-Samaan** [comment]
Deployed on: 
https://curious-florentine-7b2e32.netlify.app/

File: week1/project/ecommerce/public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "short_name": "React App",
+  "name": "Create React App Sample",
+  "icons": [
+    {
+      "src": "favicon.ico",
+      "sizes": "64x64 32x32 24x24 16x16",
+      "type": "image/x-icon"
+    },
+    {
+      "src": "logo192.png",
```
Comment (crevulus) [inline week1/project/ecommerce/public/manifest.json:11]: You have an error in your browser console because the site cannot find these image files 🟢

File: week1/project/ecommerce/src/Category.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import categories from "./fake-data/all-categories.js";
+
+export default function Category({selectCategory, categoryId}) {
+  return (
+    <div className='category-container'>
+    {categories.map((category, index) => {
+      return <div
```
Comment (crevulus) [inline week1/project/ecommerce/src/Category.js:8]: This file is a bit hard to read because of the formatting. Please make sure your eslint extension is working properly. Try using format on save: https://geekflare.com/auto-format-in-vs-code/ 🟢

File: week1/project/ecommerce/src/Category.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+export default function Category({selectCategory, categoryId}) {
+  return (
+    <div className='category-container'>
+    {categories.map((category, index) => {
+      return <div 
+      id={index} 
+      className={
+        categoryId === index
+        ?'category-btn selected'
```
Comment (crevulus) [inline week1/project/ecommerce/src/Category.js:12]: You type out `category-btn` twice; that class is always applied to the button, so I wonder if there's a way you could type it out just once and make just the other class conditional? 🟢

File: week1/project/ecommerce/src/Category.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import categories from "./fake-data/all-categories.js";
+
+export default function Category({selectCategory, categoryId}) {
+  return (
+    <div className='category-container'>
+    {categories.map((category, index) => {
```
Comment (crevulus) [inline week1/project/ecommerce/src/Category.js:7]: Don't use `div` when you need a clickable element! html elements come with a lot of useful functionality and some accessibility features too -- `button` was made to be clicked, so make sure you use that (or something else that is intended for interactivity) 🟠

File: week1/project/ecommerce/src/App.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import allCategories from "./fake-data/all-categories";
+import allProducts from "./fake-data/all-products"
+import Category from "./Category";
+import Products from './Products';
+
+function App() {
+  const [categoryId, setCategoryId] = useState(null)
+
+  function selectCategory(e){
+    const categoryId = e.target.id
```
Comment (crevulus) [inline week1/project/ecommerce/src/App.js:12]: `id` is fine, but you will see `value` used a lot more often! 🟢

File: week1/project/ecommerce/src/App.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    </div>
+  );
+}
+
+export default App;
+
+
+
+
+// // import logo from './logo.svg';
```
Comment (crevulus) [inline week1/project/ecommerce/src/App.js:29]: If you're not keeping this for any particular reason it is best to delete it before you put it into your PRs. If you are keeping it for some reason, make sure you add a comment explainaing why! 🟢

File: week1/project/ecommerce/src/App.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  const [categoryId, setCategoryId] = useState(null)
+
+  function selectCategory(e){
+    const categoryId = e.target.id
+    setCategoryId(categoryId);
+  }
+  return (
+    <div className="App">
+      <h1>Products</h1>
+      <Category selectCategory={selectCategory} categoryId={categoryId} setCategoryId={setCategoryId}/>
```
Comment (crevulus) [inline week1/project/ecommerce/src/App.js:18]: You have too many props! You aren't using `setCategoryId` in the component, so make sure you delete it. If you leave props where they're not needed, your component logic will quickly become hard to follow. 🔴

File: week1/project/ecommerce/src/Category.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      return <div 
+      id={index} 
+      className={
+        categoryId === index
+        ?'category-btn selected'
+        :'category-btn'
+      }
+      onClick={e => {
```
Comment (crevulus) [inline week1/project/ecommerce/src/Category.js:15]: Your `categoryId` and your `index` do not match, so your `selected` styled never get added. Console log out both variables and check what is different between them. 🔴

File: week1/project/ecommerce/src/App.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import './App.css'
+import React, {useState} from "react";
+import allCategories from "./fake-data/all-categories";
```
Comment (crevulus) [inline week1/project/ecommerce/src/App.js:3]: Delete unused imports 🟢

File: week1/project/ecommerce/src/App.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+import allProducts from "./fake-data/all-products"
+import Category from "./Category";
+import Products from './Products';
+
+function App() {
+  const [categoryId, setCategoryId] = useState(null)
+
+  function selectCategory(e){
```
Comment (crevulus) [inline week1/project/ecommerce/src/App.js:11]: We usually expect students to use another method to make sure the data matches, but using the id/index works too! Smart solution.

File: week1/project/ecommerce/src/ProductItem.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+const ProductItem = ({ product }) => {
+  const formattedPrice = new Intl.NumberFormat('en-GB', {
```
Comment (crevulus) [inline week1/project/ecommerce/src/ProductItem.js:4]: Nice use of this API!

File: week1/project/ecommerce/src/Products.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import products from './fake-data/all-products'
+import ProductItem  from './ProductItem.js'
+
+export default function Products({ selectedCategoryId}) {
+  return (
+    <div className='products-container'>
+      {products.map((product) => {
+        let categoryId;
```
Comment (crevulus) [inline week1/project/ecommerce/src/Products.js:9]: This is an unconventional solution, but we can allow it in theory. You will, however, need to clean up the code to make it more readable. There are a couple of thing you're doing in this component, that we should generally try to avoid in React components:

1. Do as much of your logic as you can **before** your `return` statement. This is partly better for readability, and also means you don't redo the same logic over and over in your `map` method.
2. Using the "wrong" html element attributes to change logic. It's best to use `value` and other similar props rather than using `className`. `className` is for styling and applying classes. It's important to keep it that way or your components will become very messy very quickly -- you can imagine that if you're working with a large-scale application and class names are controlling both styling and javascript logic, then they will be hard to manage.

I will let you decide how to change this component, so long as you lift the logic out of the `return` statement and just return dom elements inside a `map` method in the `return` statement. Please let me know ifn you need help with that. 🔴

**Review by crevulus — COMMENTED** [review]
Hi Sally,

Good work! Especially on styling - I can see you have an eye for these things.

I have left some comments for you to work on. Your solutions are unconventional in some places, but you can still make them work in a "React" way if you change them a little. Otherwise we can talk about more simplified solutions.

File: week1/project/ecommerce/src/Category.js (hunk 8 above)
Comment (Sally-Samaan) [reply week1/project/ecommerce/src/Category.js:15]: oh yes! I knew I had a problem here but couldn’t see it

File: week1/project/ecommerce/src/Products.js (hunk 12 above)
Comment (Sally-Samaan) [reply week1/project/ecommerce/src/Products.js:9]: thank you for the comments and the instructions :))
I will work on them as soon as i can

**Comment by crevulus** [comment]
@Sally-Samaan I don't see any changes to these files - can you please handle these comments?
